foo.cpp

```diff
@@ -1,12 +1,36 @@
 #include <cassert>
 #include <cstdint>
+#include <cstdlib> // abort
+
+#if defined(__clang_analyzer__)
+# define my_assert(A) \
+  do { \
+    if (!(A)) { \
+      __builtin_unreachable(); \
```

Comment:
@isaachier Did the trick.

Reply:
Nice. I was thinking of doing something like that. Upstream might not be interested in that change but we can do it in hunter-packages.

Reply:
May be related, another feature: https://stackoverflow.com/a/18230740/2288008